tests/integration/targets/cargo/tasks/setup.yml

```
@@ -26,3 +26,17 @@
       has_cargo: true
   when:
     - ansible_system == 'FreeBSD' and ansible_distribution_version is version('13.0', '>')
+
+- block:
+  - name: Download rustup
+    get_url:
+      url: https://sh.rustup.rs
+      dest: /tmp/sh.rustup.rs
+      mode: "0750"
+      force: true
+  - name: Install rustup cargo
+    shell: /tmp/sh.rustup.rs -y
```

Sorry, I missed this. `shell` should only ever used if you really need a shell, to run a command always use `command`:
```suggestion
    command: /tmp/sh.rustup.rs -y
```
(Ref: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/shell_module.html#notes)